pilot/proxy/envoy/header.go

```diff
@@ -58,6 +59,59 @@ func buildHTTPRouteMatch(matches *routing.MatchCondition) *HTTPRoute {
 	}
 }
 
+func buildHTTPRouteMatches(matches []*routingv2.HTTPMatchRequest) []*HTTPRoute {
+	if len(matches) == 0 {
+		return []*HTTPRoute{{Prefix: "/"}}
+	}
+
+	routes := make([]*HTTPRoute, 0, len(matches))
+	for _, match := range matches {
+		route := &HTTPRoute{}
```

Comment:
I think this can be refactored in a cleaner way. iterate through list of routes, and for each route, iterate through list of matches, and for each match, run buildHttpRoute()..